api-put-object.go

```diff
@@ -299,11 +300,17 @@ func (c *Client) putObjectMultipartStreamNoLength(ctx context.Context, bucketNam
 	if err != nil {
 		return UploadInfo{}, err
 	}
+
+	if !opts.SendContentMd5 {
+		opts.UserMetadata = map[string]string{"X-Amz-Checksum-Algorithm": "CRC32C"}
```

Comment:
I think you should not initialize a new UserMetadata here if opts.userMetadata is not nil, but just only add if it is already initialized.

This is called just before newUploadID() and usually user metadata are initialized in this stage, NewMultipartUpload().